### ISSUE TITLE
GCC fails to build on 10.4 Intel

### ISSUE THREAD

**Issue by mistydemeo** [mistydemeo/tigerbrew#471]
GCC 5.4 fails to build due to a missing constant for `sysconf`:

```
libtool: compile:  /private/tmp/gcc20170403-22127-15tuf13/gcc-5.2.0/build/./gcc/xgcc -B/private/tmp/gcc20170403-22127-15tuf13/gcc-5.2.0/build/./gcc/ -B/usr/local/Cellar/gcc/5.2.0/i686-apple-darwin8.11.1/bin/ -B/usr/local/Cellar/gcc/5.2.0/i686-apple-darwin8.11.1/lib/ -isystem /usr/local/Cellar/gcc/5.2.0/i686-apple-darwin8.11.1/include -isystem /usr/local/Cellar/gcc/5.2.0/i686-apple-darwin8.11.1/sys-include "-DPACKAGE_NAME=\"Cilk Runtime Library\"" -DPACKAGE_TARNAME=\"cilk-runtime-library\" -DPACKAGE_VERSION=\"2.0\" "-DPACKAGE_STRING=\"Cilk Runtime Library 2.0\"" -DPACKAGE_BUGREPORT=\"cilk@intel.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"cilk-runtime-library\" -DVERSION=\"2.0\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_ALLOCA_H=1 -DHAVE_ALLOCA=1 -DHAVE_ATTRIBUTE_VISIBILITY=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -I../../../libcilkrts -I../../../libcilkrts/include -I../../../libcilkrts/runtime -I../../../libcilkrts/runtime/config/x86 -DIN_CILK_RUNTIME=1 -pthread -fcilkplus -I../../../libcilkrts/include -I../../../libcilkrts/runtime -I../../../libcilkrts/runtime/config/x86 -DIN_CILK_RUNTIME=1 -fcilkplus -std=c99 -g -O2 -Os -w -pipe -march=prescott -mmacosx-version-min=10.4 -MT sysdep-unix.lo -MD -MP -MF .deps/sysdep-unix.Tpo -c ../../../libcilkrts/runtime/sysdep-unix.c  -fno-common -DPIC -o .libs/sysdep-unix.o
../../../libcilkrts/runtime/sysdep-unix.c: In function 'write_version_file':
../../../libcilkrts/runtime/sysdep-unix.c:736:52: error: '_SC_NPROCESSORS_ONLN' undeclared (first use in this function)
     fprintf(fp, "System cores: %d\n", (int)sysconf(_SC_NPROCESSORS_ONLN));
                                                    ^
../../../libcilkrts/runtime/sysdep-unix.c:736:52: note: each undeclared identifier is reported only once for each function it appears in
```

This was reported upstream, but hasn't been resolved in any upstream release: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=64184 As far as I see, this is an Intel-only issue, which is why we haven't encountered it yet.

This is similar to the issue in #183.

cc @ranma42 - this may interest you. The other GCC issues I've encountered are being tracked/fixed in PR #470.

**Comment by mistydemeo** [comment]
Strangely, like I mentioned in #183, `_SC_NPROCESSORS_ONLN` is documented in the `sysconf` manpage but is missing from the headers.

**Comment by mistydemeo** [comment]
Fixed with the latest commit in #470.
